### PR TITLE
Allow test to depend on executable target in cross build. 

### PR DESCRIPTION
**Pull request by FredericHamel**
Allow mixed test in crossbuild.
Use exe.is_cross to check if the executable tested
need to be wrapped by the exe_wrapper.

This option test is added in mesonbuild/backend/backends.py:write_test_serialisation.

### CODE REVIEW

File: mesonbuild/backend/backends.py [hunk 1]
```diff
@@ -506,6 +506,8 @@ def write_test_serialisation(self, tests, datafile):
             is_cross = self.environment.is_cross_build() and \
                 self.environment.cross_info.need_cross_compiler() and \
                 self.environment.cross_info.need_exe_wrapper()
+            if isinstance(exe, (build.CustomTarget, build.BuildTarget)):
+              is_cross = is_cross and exe.is_cross
```
Comment (ignatenkobrain) [inline mesonbuild/backend/backends.py]: __[flake8]__
* [E111] indentation is not a multiple of four

Comment (FredericHamel) [inline mesonbuild/backend/backends.py]: My mistake. Thank you for mentionning.

**Comment by FredericHamel** [comment]
1. I removed the merge commit, was my mistake. The bug fixed by this is simply enable cross testing alongside native testing. 
2. When adding tests in a cross build configuration, all executable are execute using the host configuration. For example, when cross testing windows executable alongside elf64 executable on linux, all the tests would be run using wine (the exe_wrapper) even though they are native linux executable. The patch check if the executable was cross build or not.
3. I might not have seen that is_cross attribute wasn't part of CustomTargets.

**Comment by jpakkane** [comment]
The history is a bit confusing. The first commit changes some code which the second commit reverts immediately afterwards. Is the current status (only add the is_cross check for build target) the exact change you wish to have?

**Comment by FredericHamel** [comment]
The only change is only this. This allow the test to be correctly execute depending whether it is cross build or not.

**Comment by jpakkane** [comment]
Looks good, but this needs a test. Add something like this at the end of `common tests/1 trivial/meson.build`

    if meson.is_cross()
      native_exe = executable('native-trivialprog', sources : sources)
      test('native exe in cross build', native_exe)
    endif

File: test cases/common/1 trivial/meson.build [hunk 2]
```diff
@@ -11,6 +11,11 @@ if meson.get_compiler('c').get_id() == 'intel'
   add_project_arguments('-diag-error', '10159', language : 'c')
 endif
 
+if meson.is_cross_build()
+  native_exe = executable('native-trivialprog', sources : sources)
```
Comment (jpakkane) [inline test cases/common/1 trivial/meson.build]: Sorry, this should have had `native : true` in it.

Comment (FredericHamel) [inline test cases/common/1 trivial/meson.build]: I noticed after doing a push... I am testing the test locally before pushing.

**Comment by nirbheek** [comment]
The changes look good, but can you please squash all this into one commit? There's a bunch of noise because of changes that you later revert or add to. :)

**Comment by jpakkane** [comment]
No need, I'll use Github's "squash into one and merge" button. Thanks.

**Pull request merged**